### PR TITLE
BUG: alloc_tensor offset and reshape shape should be on the CPU

### PR DESCRIPTION
**Pull request by mbs-octoml**
The VM ManifestAlloc pass was allocating constants in a few places I
forgot to tag with on_device for the host/cpu. As a result the runtime
would (silently) do the x-device copy, which destroys perf.

To make this easier to spot in the future added a 'constants' property
to the VM Executable to dump the shape & device for all VM constants.

This is CORE-102 in OctoML JIRA.

(Special thanks to @mbrookhart  & @tkonolige for finding this.)

### CODE REVIEW

**Comment by mbs-octoml** [comment]
With this fix the ONNX perf test suite agrees in timing to 2 decimal places, and the vm executables agree exactly.

File: tests/python/relay/test_vm.py [hunk 1]
```diff
@@ -999,6 +999,71 @@ def test_shape_func_nested_function():
...
+            }
+        """
+        )
+
+    exe = relay.vm.compile(
+        input(),
+        tvm.target.Target("cuda"),
+    )
+
+    print(exe.constants)
```
Comment (mbrookhart) [inline tests/python/relay/test_vm.py:1024]: Nit, remove prints in tests

File: tests/python/relay/test_vm.py [hunk 2]
```diff
@@ -999,6 +999,71 @@ def test_shape_func_nested_function():
...
```
Comment (mbrookhart) [inline tests/python/relay/test_vm.py:1059]: Nit, remove prints in tests

**Review by mbrookhart — APPROVED** [review]
LGTM, a couple of nits

File: tests/python/relay/test_vm.py (hunk 1 above)
Comment (mbs-octoml) [reply tests/python/relay/test_vm.py:1024]: done

File: tests/python/relay/test_vm.py (hunk 2 above)
Comment (mbs-octoml) [reply tests/python/relay/test_vm.py:1059]: done

**Comment by junrushao1994** [comment]
Thanks for the fix!!

**Pull request merged**